packages/store-ui/src/atoms/Slider/Slider.tsx

```diff
@@ -0,0 +1,150 @@
+  testId = 'store-slider',
+  showValues = true,
+  ariaLabel,
+  getAriaValueText,
+}: SliderProps) => {
+  const [minVal, setMinVal] = useState(min)
+  const [maxVal, setMaxVal] = useState(max)
+
+  const minValRef = useRef(min)
+  const maxValRef = useRef(max)
+  const range = useRef<HTMLDivElement>(null)
+
+  const getPercent = useCallback(
+    (value) => Math.round(((value - min) / (max - min)) * 100),
+    [min, max]
+  )
+
+  // width of the range to reduce from the left side
+  useEffect(() => {
+    const minPercent = getPercent(minVal)
+    const maxPercent = getPercent(maxValRef.current)
+
+    if (range.current) {
+      range.current.style.left = `${minPercent}%`
+      range.current.style.width = `${maxPercent - minPercent}%`
+    }
+  }, [minVal, getPercent])
+
+  // width of the range to reduce from the right side
+  useEffect(() => {
+    const minPercent = getPercent(minValRef.current)
+    const maxPercent = getPercent(maxVal)
+
+    if (range.current) {
+      range.current.style.width = `${maxPercent - minPercent}%`
+    }
+  }, [maxVal, getPercent])
+
+  useEffect(() => {
+    onChange?.({ min: minVal, max: maxVal })
+  }, [minVal, maxVal, onChange])
+
+  return (
+    <div
+      data-store-slider
+      data-testid={testId}
+      aria-label={ariaLabel}
+      // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
+      role="slider"
+    >
+      <div data-store-slider-range>
+        <div ref={range} data-store-slider-track />
+        <input
+          type="range"
+          min={min}
+          max={max}
+          value={minVal}
+          onChange={(event) => {
+            const value = Math.min(Number(event.target.value), maxVal - 1)
+
+            setMinVal(value)
+            minValRef.current = value
+          }}
+          data-store-slider-thumb="left"
+          style={{ zIndex: minVal > max - 100 ? 5 : 'auto' }}
+          aria-valuemin={min}
+          aria-valuemax={max}
+          aria-valuenow={minVal}
+          aria-label={String(minVal)}
+          aria-labelledby={
+            getAriaValueText ? getAriaValueText(minVal, 'min') : undefined
+          }
+        />
+        <input
+          type="range"
+          min={min}
+          max={max}
+          value={maxVal}
+          onChange={(event) => {
+            const value = Math.max(Number(event.target.value), minVal + 1)
+
+            setMaxVal(value)
+            maxValRef.current = value
+          }}
+          data-store-slider-thumb="right"
+          aria-valuemin={min}
+          aria-valuemax={max}
+          aria-valuenow={maxVal}
+          aria-label={String(maxVal)}
+          aria-labelledby={
+            getAriaValueText ? getAriaValueText(maxVal, 'max') : undefined
+          }
+        />
+      </div>
+      {showValues && (
+        <div data-store-slider-values>
+          <div data-store-slider-value="min">{minVal}</div>
+          <div data-store-slider-value="max">{maxVal}</div>
+        </div>
```

Comment:
I don't think we should include this feature. It adds unnecessary behaviors to the atom and I think it deviates from the main goal of the atom. Users of this atom can accomplish the same thing by using the onChange callback and implementing the min/max values themselves.

I agree it's a nice-to-have feature, but I think this shouldn't be on the atom itself. For example, we already see the need to add a formatter function so people could treat it as prices. For that, we can easily create a molecule that combines this atom without this feature and the price atom to accomplish the same thing, but even more composable.

Reply:
Just adding my opinion to this, I agree with Ícaro. The `Slider` could be just the range selector, without these labels. Then, a molecule such as the `Price Range` would use this atom + the `Price` atom.

Reply:
hm, makes sense. So, I will convert that to an issue.

Reply:
@victorhmp and @icazevedo can you review? https://github.com/vtex/faststore/pull/924